pkg/loop/internal/relayer/pluginprovider/chainreader/chain_reader.go

```diff
@@ -228,6 +243,20 @@ func (c *Server) GetLatestValue(ctx context.Context, request *pb.GetLatestValueR
 	return &pb.GetLatestValueReply{RetVal: encodedRetVal}, nil
 }
 
+func (c *Server) BatchGetLatestValues(ctx context.Context, pbRequest *pb.BatchGetLatestValuesRequest) (*pb.BatchGetLatestValuesReply, error) {
+	request, err := convertBatchGetLatestValuesRequestFromProto(pbRequest, c.impl)
+	if err != nil {
+		return nil, err
+	}
+
+	reply, err := c.impl.BatchGetLatestValues(ctx, request)
+	if err != nil {
+		return nil, err
+	}
+
+	return newPbBatchGetLatestValuesReply(reply, c.encodeWith)
```

Comment:
so if one of the embedded calls results in an error, this returns no error. is that what user would expect? i kind of doubt it.

every caller will have to loop thru all the of result and check every string Err for a value. 

would it be easier to use if this returns an errors.Join of all the accumulated inner errors or something similar? 

my main concern is whether or not callers will correctly check for individual errors. perhaps really good documentation is sufficient, but that seems filmsy

Reply:
This is how our current implementation of batch call works [here](https://github.com/smartcontractkit/ccip/blob/54ee4f13143d3e414627b6a0b9f71d5dfade76c5/core/services/ocr2/plugins/ccip/internal/rpclib/evm.go#L25). Although seems like outputs are always parsed with [ParseOutputs()](https://github.com/smartcontractkit/ccip/blob/dacf09fbba3fa20e1fc596640b74f8b15d3d2b58/core/services/ocr2/plugins/ccip/internal/rpclib/evm.go#L297), which returns an error as soon as it encounters one, but it does leave room for handling this differently if you don't care if only some results error.

Probably best to ask people that will use this what is the best approach. CC @patrickhuie19 @dimkouv 

Do you want `BatchGetLatesetValues()` to return an `errors.Join` of all the accumulated errors if there are any? Or is the current approach okay where you have to loop through responses to check for errors ( which you have to do anyway to collect the results)? Or do you have something else in mind?

Reply:
Also the same parse function can be used, just without the parse part since codec does that now, which would just make it an error checking loop

Reply:
hmm.. so It's going to be used like this:

```
myBatchCalls = [...]

results, err := reader.BatchCall(myBatchCalls)
if err -> return err

for res in results:
   if res.err -> return err
   handle res.data
```

Someone has to forget the `if res.err -> return err` which might happen...

Maybe we can somehow enforce this check by doing:
```
myBatchCalls = [...]

results, err := reader.BatchCall(myBatchCalls)
if err -> return err

for res in results:
   data, err = res.data()
```

Reply:
Nice, I like the approach in the second example, I'll hide data behind a getter that checks for error

Reply:
resolved